enterprise/backend/test/metabase_enterprise/advanced_permissions/common_test.clj

```diff
@@ -66,29 +65,56 @@
                           (user-permissions :rasta)))))))))
 
 (deftest new-database-view-data-permission-level-test
+  (mt/with-additional-premium-features #{:sandboxes :advanced-permissions}
+    (mt/with-temp [:model/Database         {db-id :id}      {}
+                   :model/PermissionsGroup {group-id :id}   {}]
+      (let [perm-value (fn [db-id] (t2/select-one-fn :perm_value
+                                                     :model/DataPermissions
+                                                     :db_id     db-id
+                                                     :group_id  group-id
+                                                     :table_id  nil
+                                                     :perm_type :perms/view-data))]
+        (testing "A new database defaults to `:unrestricted` if no other perms are set"
+          (mt/with-temp [:model/Database {db-id-2 :id} {}]
```

Comment:
Refactored this test to be a little more of an integration test — creating new DBs via `with-temp` instead of just calling `new-database-view-data-permission-level` in isolation